hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -181,4 +213,136 @@ default Table newHmsTable(String hmsTableOwner) {
 
     return newTable;
   }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  default void commitWithLocking(
```

Comment:
Idea here is to generalise doCommit from [HiveTableOperations](https://github.com/apache/iceberg/blob/53a1c8671dd1b9b93f4a857230008c812d79ddbf/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L171). It should be used by view too.